src/coordinator/tsdb/remote/compressed_codecs.go

```diff
@@ -142,25 +162,55 @@ func CompressedSeriesFromSeriesIterator(it encoding.SeriesIterator) (*rpc.Series
 		})
 	}
 
-	tags, err := compressedTagsFromTagIterator(it.Tags())
-	if err != nil {
-		return nil, err
-	}
-
 	start := xtime.ToNanoseconds(it.Start())
 	end := xtime.ToNanoseconds(it.End())
+	tagIter := it.Tags()
+	compressedTags, canCompress := compressedTagsFromTagIteratorWithEncoder(tagIter, iterPools)
+
+	var (
+		tags []*rpc.Tag
+		err  error
+	)
+
+	if !canCompress {
```

Comment:
Its strange to make it optional. Clients might be simpler if compressed datapoints always correspond to compressedTags as well. Clients can then just always go one path instead of switching based on error